{{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/{{cookiecutter.django_default_app_name}}/tasks.py

```diff
@@ -1,13 +1,40 @@
 {%- if cookiecutter.use_celery == 'y' -%}
 import structlog
+from celery import Task
 from celery.utils.log import get_task_logger
 
 from {{cookiecutter.django_project_name}}.celery import app
 
 logger = structlog.wrap_logger(get_task_logger(__name__))
 
 
-@app.task
+def send_to_dead_letter_queue(task: Task, exc, task_id, args, kwargs, einfo):
+    """
+    Hook to put a task into dead letter queue when it fails.
+
+    The task should be annotated with `on_failure=send_to_dead_letter_queue`.
+    Once the reason of tasks failure is fixed, the task can be re-processed
+    by running a worker on dead letter queue:
+
+        celery -A mrc.celery worker -Q dead_letter
+
+    If tasks fails again, it will be put back to dead letter queue. This can
+    potentially create an infinite loop, so make sure to fix the reason of
+    failure before re-processing the task, or stop the worker after a few
+    retries. The process of re-processing tasks is supposed to be manual,
+    thus there are no complicated mechanisms to prevent infinite loops.
```

Comment:
Shouldn't we add a dead dead letter queue for this and then the fixing processor would be configured to pick stuff up from dead_letter and run it but on failure put in dead_dead_letter, then if this happens you can fix the code in such a way that it's not sending anything to dead_dead, run through the whole dead and finally run like a special task which takes everything from dead_dead and moves to dead or runs on dead_dead and puts to dead? I doubt we'll always be perfect and there isn't anything that would warn us about the infinite loop, so putting in a process to handle it properly seems practical to me. It's a good moment to think about how to handle it.

Also this sounds like something that should be described somewhere in the documentation of our framework, not just in this docstring here.

Reply:
I just added a command to move tasks from one queue to another. No infinite loops involved.

https://github.com/reef-technologies/cookiecutter-rt-django/commit/322fc5e89692b00a158da25ee3b9402ff9edacde